<?xml version="1.0"?>
<launch>
    <!-- <args for urdf/> -->
    <arg name="camera_model"          default="OAK-D"   /> <!-- 'zed' or 'zedm' -->
    <arg name="tf_prefix"             default="oak"         />
    <arg name="base_frame"            default="oak-d_frame" />
    <arg name="parent_frame"          default="oak-d-base-frame" />
    <arg name="publisher_name"        default="replay_publisher" />

    <arg name="cam_pos_x"             default="0.0" /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_pos_y"             default="0.0" /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_pos_z"             default="0.0" /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_roll"              default="0.0" /> <!-- Orientation respect to base frame (i.e. "base_link) -->
    <arg name="cam_pitch"             default="0.0" /> <!-- Orientation respect to base frame (i.e. "base_link) -->
    <arg name="cam_yaw"               default="0.0" /> <!-- Orientation respect to base frame (i.e. "base_link) -->

    <arg name="camera_param_uri"      default="package://depthai_examples/params/camera" />
    <arg name="subpixel"              default="true"/>
    <arg name="confidence"            default="200" /> 
    <arg name="LRchecktresh"          default="5"   /> 
    <arg name="lrcheck"               default="false" /> 
    <arg name="extended"              default="false" /> 
    <arg name="previewWidth"          default="416" />
    <arg name="previewHeight"         default="416" />
    <arg name="rgbWidth"              default="1920" />
    <arg name="rgbHeight"             default="1080" />
    <arg name="monoWidth"             default="1280" />
    <arg name="monoHeight"            default="720" />
    <arg name="rgbScaleNumerator"     default="2"/>
    <arg name="rgbScaleDinominator"   default="3"/>

    <arg name="rgbVideoPath"          default="$(find oakd)/scripts/records/rgb.h265"/>
    <arg name="leftVideoPath"         default="$(find oakd)/scripts/records/left.mjpeg"/>
    <arg name="rightVideoPath"        default="$(find oakd)/scripts/records/right.mjpeg"/>
    <arg name="previewVideoPath"      default="$(find oakd)/scripts/records/preview.h265"/>
    
    <arg name="nnPath"                default="$(find oakd)/resources/yolov4_tiny_coco_416x416_openvino_2021.4_6shave_bgr.blob"/>



    <include file="$(find depthai_descriptions)/launch/urdf.launch">
        <arg name="base_frame"      value="$(arg  base_frame)"  />
        <arg name="parent_frame"    value="$(arg  parent_frame)"/>
        <arg name="camera_model"    value="$(arg  camera_model)"/>
        <arg name="tf_prefix"       value="$(arg  tf_prefix)" />
        <arg name="cam_pos_x"       value="$(arg  cam_pos_x)" />
        <arg name="cam_pos_y"       value="$(arg  cam_pos_y)" />
        <arg name="cam_pos_z"       value="$(arg  cam_pos_z)" />
        <arg name="cam_roll"        value="$(arg  cam_roll)"  />
        <arg name="cam_pitch"       value="$(arg  cam_pitch)" />
        <arg name="cam_yaw"         value="$(arg  cam_yaw)"   />
    </include>

    <node name="$(arg publisher_name)" pkg="oakd" type="replay_publisher_node" output="screen" required="true">
        <param name="tf_prefix"           value="$(arg tf_prefix)"     />
        <param name="subpixel"            value="$(arg subpixel)"/>
        <param name="confidence"          value="$(arg confidence)"/>
        <param name="LRchecktresh"        value="$(arg LRchecktresh)"/>
        <param name="previewWidth"        value="$(arg previewWidth)"/>
        <param name="previewHeight"       value="$(arg previewHeight)"/>
        <param name="rgbWidth"            value="$(arg rgbWidth)"/>
        <param name="rgbHeight"           value="$(arg rgbHeight)"/>
        <param name="monoWidth"           value="$(arg monoWidth)"/>
        <param name="monoHeight"          value="$(arg monoHeight)"/>
        <param name="lrcheck"             value="$(arg lrcheck)"/> 
        <param name="extended"            value="$(arg extended)"/> 
        <param name="rgbVideoPath"        value="$(arg rgbVideoPath)"/>
        <param name="leftVideoPath"       value="$(arg leftVideoPath)"/>
        <param name="rightVideoPath"      value="$(arg rightVideoPath)"/>
        <param name="previewVideoPath"    value="$(arg previewVideoPath)"/>
        <param name="nnPath"              value="$(arg nnPath)"/>
        <param name="rgbScaleNumerator"   value="$(arg rgbScaleNumerator)"/> 
        <param name="rgbScaleDinominator" value="$(arg rgbScaleDinominator)"/> 
    </node>

</launch>
